File: ployglots/mta-ployglot-pkg-fetch-client/packages/py/examples/integrations/statsig.py
"""Example: Statsig Console API.

https://docs.statsig.com/console-api/introduction

Env:
    STATSIG_HOST  — usually 'https://statsigapi.net'
    STATSIG_USER  — placeholder (Statsig uses token-only auth; kept for symmetry)
    STATSIG_PASS  — a Console API key (server secret)
    HTTPS_PROXY   — optional outbound proxy.

Endpoint exercised: GET /console/v1/feature_gates — first page of gates.

Run:
    python -m examples.integrations.statsig
"""

from __future__ import annotations

import asyncio
import sys

from fetch_http_client import APIKeyAuth, AsyncClient

from .._shared import build_proxy, optional_env, require_env


async def main() -> None:
    host = optional_env("STATSIG_HOST", "https://statsigapi.net")
    _user = optional_env("STATSIG_USER", "")  # kept for ENV symmetry
    token = require_env("STATSIG_PASS")

    proxy = build_proxy({})

    kwargs: dict = {
        "base_url": host,
        # Statsig Console API uses the STATSIG-API-KEY header.
        "auth": APIKeyAuth(token, "STATSIG-API-KEY"),
        "headers": {"accept": "application/json"},
    }
    if proxy is not None:
        kwargs["proxy"] = proxy

    async with AsyncClient(**kwargs) as client:
        resp = await client.get("/console/v1/feature_gates")
        resp.raise_for_status()
        data = resp.json()
        gates = data.get("data", [])
        sys.stdout.write(f"Statsig {host} → {len(gates)} feature gates:\n")
        for g in gates[:5]:
            sys.stdout.write(f"  - {g.get('id')} ({g.get('name')}) enabled={g.get('isEnabled')}\n")


if __name__ == "__main__":
    try:
        asyncio.run(main())
    except Exception as err:
        sys.stderr.write(f"Statsig example failed: {err}\n")
        sys.exit(1)

File: ployglots/mta-ployglot-pkg-fetch-client/packages/ts/examples/integrations/statsig.ts
// @ts-nocheck
/**
 * Example: Statsig Console API (public docs: https://docs.statsig.com/console-api/introduction).
 *
 * Env:
 *   STATSIG_HOST  — usually 'https://statsigapi.net'
 *   STATSIG_USER  — placeholder (kept for ENV symmetry; Statsig uses token-only auth)
 *   STATSIG_PASS  — a Console API key (server secret)
 *   HTTPS_PROXY   — optional outbound proxy.
 *
 * Endpoint exercised: GET /console/v1/feature_gates — first page of gates in the project.
 *
 * Run: tsx examples/integrations/statsig.ts
 */
import { AsyncClient, APIKeyAuth } from '@polyglot/fetch-http-client';
import { buildProxy, optionalEnv, requireEnv } from '../_shared.js';

interface FeatureGate {
  id: string;
  name: string;
  isEnabled: boolean;
}

async function main(): Promise<void> {
  const host = optionalEnv('STATSIG_HOST', 'https://statsigapi.net');
  const _user = optionalEnv('STATSIG_USER', '');
  const token = requireEnv('STATSIG_PASS');

  const proxy = buildProxy({});

  const client = new AsyncClient({
    baseUrl: host,
    // Statsig Console API uses the STATSIG-API-KEY header.
    auth: new APIKeyAuth(token, 'STATSIG-API-KEY'),
    headers: { accept: 'application/json' },
    ...(proxy ? { proxy } : {}),
  });

  try {
    const resp = await client.get('/console/v1/feature_gates');
    resp.raiseForStatus();
    const data = await resp.json<{ data: FeatureGate[] }>();
    process.stdout.write(`Statsig ${host} → ${data.data.length} feature gates:\n`);
    for (const g of data.data.slice(0, 5)) {
      process.stdout.write(`  - ${g.id} (${g.name}) enabled=${g.isEnabled}\n`);
    }
  } finally {
    await client.close();
  }
}

main().catch((err: unknown) => {
  process.stderr.write(`Statsig example failed: ${(err as Error).message}\n`);
  process.exitCode = 1;
});
